analyses/molecular-subtyping-HGG/06-HGG-molecular-subtyping-gene-expression.Rmd

```diff
@@ -0,0 +1,173 @@
+
+# File path to results directory
+results_dir <-
+  file.path(root_dir, "analyses", "molecular-subtyping-HGG", "results")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+### Read in Files
+
+The gene expression files were generated by filtering z-scored expression data
+(handling polyA selection and stranded expression files separately) for the
+purpose of subtyping HGG samples. 
+
+```{r}
+# Read in subsetted polyA expression data
+polya_expression_df <-
+  read_rds(file.path(input_dir, "hgg_zscored_expression.polya.RDS"))
+
+# Read in subsetted stranded expression data
+stranded_expression_df <-
+  read_rds(file.path(input_dir, "hgg_zscored_expression.stranded.RDS"))
+```
+
+### Output Files
+
+```{r}
+# File path to cleaned polyA expression output
+polya_output_file <- file.path(results_dir, "HGG_cleaned_expression.polya.tsv")
+
+# File path to cleaned stranded expression output
+stranded_output_file <-
+  file.path(results_dir, "HGG_cleaned_expression.stranded.tsv")
+```
+
+## Which genes' expression values are we interested in?
+
+There are three genes of interest according to [`AlexsLemonade/OpenPBTA-analysis#249`](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249).
+Specifically:
+
+* _OLIG2_ and _FOXG1_ should be highly expressed in the IDH mutant HGG subtype.
+* _TP73-AS1_ methylation and downregulation cooccur with _TP53_ mutations in the
+IDH mutant subtype.
+
+## Filter and Wrangle Expression Data
+
+```{r}
+filter_wrangle_expression <-
+  function(cleaned_expression, output_file_path) {
+    # Given an expression data.frame filtered to include only samples that meet
+    # the HGG classification criteria, select the genes of interest. It displays
+    # and writes the cleaned and filtered expression table to the given output
+    # file path.
+    
+    filtered_expression <- cleaned_expression %>%
+      as.data.frame() %>%
+      rownames_to_column("Kids_First_Biospecimen_ID") %>%
+      select(Kids_First_Biospecimen_ID,
+             FOXG1_expression_zscore = FOXG1,
+             OLIG2_expression_zscore = OLIG2,
+             TP73_AS1_expression_zscore = `TP73-AS1`) %>%
+      distinct()
+    
+    # Save expression data to file
+    write_tsv(filtered_expression, output_file_path)
+    
+    # Display `filtered_expression`
+    filtered_expression
+  }
+
+# Run `filter_wrangle_expression` on poly-A cleaned expression data
+filter_wrangle_expression(polya_expression_df , polya_output_file)
+
+# Run `filter_wrangle_expression` on stranded cleaned expression data 
+filter_wrangle_expression(stranded_expression_df , stranded_output_file)
+```
+
+## Density Plots
+
+```{r}
+plot_expression_density <- function(expression_file) {
+  # Given the cleaned expression file from the previous chunck, generate a
+  # density plot for each of the three genes of interest. This function returns
+  # three density plots(one per gene).
+  
+  # Read in file
+  expression_df <- read_tsv(expression_file)
```

Comment:
Is there a particular reason you've made it re-read in the data rather than just accept a data.frame object that is already in the environment?

Reply:
No particular reason other than that I had not assigned the cleaned expression `data.frame` in the environment, just saved it to file but I will make the change so that it is more straightforward.